components/movementsensor/imuwit/imu.go

```diff
@@ -228,6 +227,7 @@ func newWit(
 }
 
 func (imu *wit) startUpdateLoop(ctx context.Context, portReader *bufio.Reader, logger golog.Logger) {
+	imu.hasMagnetometer = false
```

Comment:
quick q - does this need to be behind a lock for reconfiguring?

Reply:
ah no it's fine, always rebuild and nothing is under lock in the update loop until parseline anyway. You can actually omit this and only set it to true in the loop - it's default false.

Reply:
but might be good to leave it in - in the unlikely case where you switch out imus without restarting, this will give you zeroes forever. But I think that's fine.

Reply:
No, startUpdateLoop is only called once per instance of wit. So unless we hit the true statement in parser, imu.hasMagnetometer is going to stay false.